ydb/core/cms/node_checkers.h

```diff
@@ -47,15 +47,25 @@ class INodesChecker {
  * Base class for simple nodes counter with some limits
  */
 class TNodesCounterBase : public INodesChecker {
+public:
+    using TNodeToPileMap = THashMap<ui32, ui32>;
+    using TNodeToPileMapPtr = std::shared_ptr<TNodeToPileMap>;
+
 protected:
+    TNodeToPileMapPtr NodeIdToPileId = std::make_shared<TNodeToPileMap>();
     THashMap<ui32, ENodeState> NodeToState;
-    ui32 LockedNodesCount;
-    ui32 DownNodesCount;
+    THashMap<ui32, ui32> LockedNodesCount; // {key: PileId; value: LockedNodesCount}
+    THashMap<ui32, ui32> DownNodesCount; // {key: PileId; value: DownNodesCount}
+
+    ui32 GetPileId(ui32 nodeId) const;
+    ui32 GetLockedNodesCount(ui32 pileId) const;
+    ui32 GetDownNodesCount(ui32 pileId) const;
 
 public:
-    TNodesCounterBase()
-        : LockedNodesCount(0)
-        , DownNodesCount(0)
+    TNodesCounterBase() {}
```

Comment:
```suggestion
    TNodesCounterBase() = default();
```
Или вообще удалить, если можно обойтись без него.

Reply:
Поправил. Вызывается этот конструктор, если кластер не в `Bridge Mode`.